### PR TITLE
Blog post for error codes

### PR DESCRIPTION
**Pull request by zpao**
Mostly just took what @keyanzhang sent me and tweaked a couple sentences/links.

screen shot 2016-07-08 at 16 20 19-fullpage

### CODE REVIEW

File: docs/_posts/2016-07-08-introducing-reacts-error-code-system.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
...
+
+In order to make debugging in production easier, we're introducing an Error Code System in 15.2.0. We developed a [gulp script](https://github.com/facebook/react/blob/master/scripts/error-codes/gulp-extract-errors.js) that collects all of our `invariant` error messages and folds them to a [JSON file](https://github.com/facebook/react/blob/master/scripts/error-codes/codes.json), and at build-time Babel uses the JSON to [rewrite](https://github.com/facebook/react/blob/master/scripts/error-codes/dev-expression-with-codes.js) our `invariant` calls in production to reference the corresponding error IDs. Now when things go wrong in production, the error that React throws will contain a URL with an error ID and relevant information. The URL will point you to a page in our documentation where the original error message gets reassembled. While we hope you don't see errors often, you can see how it works [here](/react/docs/error-decoder.html?invariant=109&args[]=Foo). This feature shouldn't require any changes on your side — use the `min.js` files in production or bundle your application code with `process.env.NODE_ENV === 'production'` and you should be good to go!
```
Comment (zpao) [inline docs/_posts/2016-07-08-introducing-reacts-error-code-system.md]: > 15.2.0

Going to turn that into a link to https://github.com/facebook/react/releases/tag/v15.2.0.

File: docs/_posts/2016-07-08-introducing-reacts-error-code-system.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+title: "Introducing React's Error Code System"
+author: keyanzhang
+---
+
+Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire. Prior to this release, we stripped out error messages at build-time and this is why you see the "Minified exception occurred; use the non-minified dev environment for the full error message and additional helpful warnings." message in production.
```
Comment (spicyj) [inline docs/_posts/2016-07-08-introducing-reacts-error-code-system.md]: Can we break it out into a blockquote to aid skimming?

---

this is why you might have seen this message in production:

> Error: Minified exception occurred; use the non-minified dev environment for the full error message and additional helpful warnings.

File: docs/_posts/2016-07-08-introducing-reacts-error-code-system.md (hunk 1 above)
Comment (spicyj) [inline docs/_posts/2016-07-08-introducing-reacts-error-code-system.md]: Let's break out a blockquote here too:

---

Now you'll see an error like this:

> Minified React error #109; visit https://facebook.github.io/react/docs/error-decoder.html?invariant=109&args[]=Foo for the full message or use the non-minified dev environment for full errors and additional helpful warnings.

Comment (spicyj) [inline docs/_posts/2016-07-08-introducing-reacts-error-code-system.md]: Maybe mention how codes keep the bundle size small? Not sure if that's implicit.

**Comment by zpao** [comment]
Updated screenshot: screen shot 2016-07-11 at 16 24 01-fullpage

**Comment by keyanzhang** [comment]
Thanks for helping me to polish this @zpao @spicyj!

**Comment by spicyj** [comment]
Can you add a paragraph break before "Prior to this release" and before "While we hope"? Sorta wall-of-text–y right now.

**Pull request merged**